### PR TITLE
Fix checkbox array and radio primitives

### LINKED ISSUE #1278: [0.8.0–0.10.11] BFormRadioGroup input & change events not working with boolean values
BFormRadioGroup with a boolean v-model is broken since 0.8.0 (still broken in 0.9.23). Was working in 0.7.3. 

Seems to be the same as #1022, which was fixed for _0_ values (#1023) but not for booleans.

Example: https://codesandbox.io/s/bootstrap-vue-next-b-form-radio-group-not-working-forked-whsl62

### LINKED ISSUE #1382: Checkbox & radio issues
If you use array as v-model checkbox overwrites the array to value (ie [] -> 1).

just test on playground.
https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/130606/780c5faf-3f55-4187-b2bf-f416e4be5b2a

Or documentation: https://bootstrap-vue-next.github.io/bootstrap-vue-next/docs/components/form-checkbox.html#checkbox-values-and-v-model

Worked on 0.7.x, just upgraded to 0.10 and many things have changed for the worst. Will keep updating the bug reports.

### PR DESCRIPTION
**Pull request by xvaara**
closes #1382 
closes #1278 

The unchecked value caused a little headache, but hopefully this is ok.

### CODE REVIEW

**Comment by xvaara** [comment]
@VividLemon do you see any problems with this one?

File: packages/bootstrap-vue-next/src/components/BFormCheckbox/BFormCheckbox.vue [hunk 1]
```diff
@@ -146,27 +146,46 @@ const localValue = computed({
...
+        }
+      }
+    }
     nextTick(() => {
       emit('change', updateValue)
     })
   },
 })
 
-watch(modelValue, (newValue) => {
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BFormCheckbox/BFormCheckbox.vue:161]: What happens if the modelValue is modified externally. How does parentData get informed of that change? In the parent component, lets say there is a toggle button that changes the modelValue. If it's in a group, how is the group informed that a child has changed?

File: packages/bootstrap-vue-next/src/components/BFormRadio/BFormRadio.vue [hunk 2]
```diff
@@ -120,30 +120,23 @@ const localValue = computed({
...
-watch(
-  () => parentData?.modelValue.value,
-  (newValue) => {
-    const isEqual = JSON.stringify(newValue) === JSON.stringify(props.value)
-    if (isEqual === true) return
-    localValue.value = false
-  }
-)
-
-watch(modelValue, (newValue) => {
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BFormRadio/BFormRadio.vue:142]: Same issue as above

**Review by VividLemon — CHANGES_REQUESTED** [review]
This isn't to say that that these are issues, but they seem to be issues. I went through many initial designs when trying out this, and there are a few syncing issues. You not only have to deal with the traditional parent-child, but you also have to deal with coordinating with that group for those parent-child changes.

File: packages/bootstrap-vue-next/src/components/BFormCheckbox/BFormCheckbox.vue (hunk 1 above)
Comment (VividLemon) [reply packages/bootstrap-vue-next/src/components/BFormCheckbox/BFormCheckbox.vue:161]:
```html
<BFormCheckboxGroup v-model="foo">
  <BFormCheckbox v-model="bar" />
</BFormCheckboxGroup>
<BButton @click="bar = !bar" />
{{foo}}
```

Comment (xvaara) [reply packages/bootstrap-vue-next/src/components/BFormCheckbox/BFormCheckbox.vue:161]: That seems weird way to use the components didn't even think about that. 😄

Comment (xvaara) [reply packages/bootstrap-vue-next/src/components/BFormCheckbox/BFormCheckbox.vue:161]: So when bar changes, should checkbox (if true) search foo for the old value, remove it and add new value? I'm not quite sure what this should do.

**Comment by xvaara** [comment]
https://stackblitz.com/edit/vitejs-vite-pqd3b7?file=src%2FApp.vue

A demo using this build

File: packages/bootstrap-vue-next/src/components/BFormCheckbox/BFormCheckbox.vue [hunk 3]
```diff
@@ -140,31 +164,73 @@ const {focused} = useFocus(input, {
...
       ? parentData.modelValue.value
           .map((el) => JSON.stringify(el))
           .includes(JSON.stringify(props.value))
+      : Array.isArray(modelValue.value)
+      ? modelValue.value?.map((el) => JSON.stringify(el)).includes(JSON.stringify(props.value))
       : JSON.stringify(modelValue.value) === JSON.stringify(props.value),
   set: (newValue) => {
-    const updateValue = newValue ? props.value : props.uncheckedValue
-
+    console.log('set', newValue, props.value, props.uncheckedValue)
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BFormCheckbox/BFormCheckbox.vue:179]: Deal with these linting warns. Either through suppressing them or fixing them (fix the 'oldValue' is defined but never used remove this, and you probably don't need the other console warn)

Comment (xvaara) [reply packages/bootstrap-vue-next/src/components/BFormCheckbox/BFormCheckbox.vue:179]: It seems my eslint is broken in my vscode :D

**Review by VividLemon — APPROVED** [review]
Assuming you've tested things and they're working, you're free to merge this in.

**Pull request merged**